tests/rptest/tests/datalake/datalake_e2e_test.py

```diff
@@ -191,14 +191,36 @@ def test_remove_expired_snapshots(self, cloud_storage_type):
                 "set tblproperties ('history.expire.max-snapshot-age-ms'='1000')"
             )
 
-            def has_one_snapshot():
+            def num_snapshots() -> int:
                 snapshots_out = spark.run_query_fetch_all(
                     f"select * from {table_name}.snapshots")
-                return len(snapshots_out) == 1
+                return len(snapshots_out)
 
-            wait_until(has_one_snapshot, timeout_sec=30, backoff_sec=1)
+            wait_until(lambda: num_snapshots() == 1,
+                       timeout_sec=30,
+                       backoff_sec=1)
             dl.wait_for_translation(self.topic_name, count)
 
+            # Externally create another snapshot. Redpanda should retain this
+            # snapshot _and_ the most recent one generated by Redpanda.
+            spark.make_client().cursor().execute(
+                f"insert into {table_name} (select * from {table_name} limit 1)"
+            )
+            num_snaps = num_snapshots()
+            assert num_snaps == 2, f"Expected 2 snapshots after writing: {num_snaps}"
+            time.sleep(10)
+            num_snaps = num_snapshots()
+            assert num_snaps == 2, f"Expected to retain 2 snapshots: {num_snaps}"
+
+            # Produce more to Redpanda. This will make the externally-created
+            # snapshot no longer the current snapshot and thus eligible for
+            # expiry. We'll still retain any Redpanda-created snapshot.
+            dl.produce_to_topic(self.topic_name, 1, 1)
+            dl.wait_for_translation(self.topic_name, count + 2)
+            wait_until(lambda: num_snapshots() == 1,
```

Comment:
q: so a tagged snapshot expires if it's superseded by another snapshot with the same tag?

Reply:
That's my understanding of it, that when we set the same tag it's "create or replace" (though it's not well documented... will add a comment somewhere)

Reply:
yeah, the rules are not very obvious from the code as such. some exposition in comments would be great 🙂